### DISCUSSION TITLE
Let's pick a date for activating TAPI (WIP-00014)

### DISCUSSION THREAD

**Discussion by aesedepece** [witnet/witnet-rust#1988] [Q&A] [answered]
[WIP-0014](https://github.com/witnet/WIPs/blob/master/wip-0014.md), aka **_TAPI_** introduces a mechanism for enabling risk-mitigated protocol upgrades by allowing block producers to signal support for specific protocol improvements, so that activation of those can automatically happen right after a supermajority of the network has signaled support for those rules for a certain period of time.

WIP-0014 itself needs to be enacted by the traditional _"will enter into force on this specific epoch"_ method.

I'm working on an activation plan for WIP-0014. What would a suitable date for the activation be?

**Comment by aesedepece** [comment, marked as answer]
Does Tuesday, July 13th sound good to you?

Under a hypothetical joint activation of WIP-0014 and WIP-0016, nodes would need to upgrade before July 27th to prevent any potential hard-forking risk.

**Reply by drcpu-github** [reply]
Is there a finalized (tagged) version for this already? Only tag I can find is tapi-rc1 which is rather old?

**Reply by aesedepece** [reply]
This is the PR porting TAPI into master: https://github.com/witnet/witnet-rust/pull/1987

**Reply by aesedepece** [reply]
And this is the PR with my activation proposal: https://github.com/witnet/WIPs/pull/62

**Reply by drcpu-github** [reply]
Yes, but that does not indicate all the changes there will be to master once a final 1.3.0 version is created, making it impossible to judge the amount of necessary upgrade work.
